source/decryptor/decryptor.c

```diff
@@ -113,7 +132,74 @@ u32 DecryptTitlekeys(void)
     }
     FileClose();
 
-    Debug("Done!");
+    return 0;
+}
+
+u32 DecryptTitlekeysNand(void)
+{
+    PartitionInfo* ctrnand_info = &(partitions[(GetUnitPlatform() == PLATFORM_3DS) ? 5 : 6]);
+    u8* buffer = BUFFER_ADDRESS;
+    EncKeysInfo *info = (EncKeysInfo*) 0x20316000;
+    u32 nKeys = 0;
+    u8* titlekey;
+    u8* titleId;
+    u32 commonKeyIndex;
+    u32 offset;
+    u32 size;
```

Comment:
This isn't C89, you don't have to declare all variables at the top of their scope. It's much better to declare the variables closest to where you're going to use them.